test/desdemona/functions/sample_functions_test.clj

```diff
@@ -1,13 +1,45 @@
 (ns desdemona.functions.sample-functions-test
-  (:require [clojure.test :refer [deftest is]]
-            [desdemona.functions.sample-functions :refer [transform-segment-shape prepare-rows]]))
+  (:require [clojure.test :refer [deftest is are]]
+            [clojure.java.io :as io]
+            [byte-streams :as bs]
+            [cheshire.core :as json]
+            [desdemona.utils :refer [kwify-map]]
+            [desdemona.functions.sample-functions :refer [prepare-rows message-origin add-message-origin build-row add-original-wrapper]]))
 
-(deftest transform-segment-shape-test
-  (let [got (transform-segment-shape {"line" [:line]} {:line "this is a log line"})
-        expected {"line" "this is a log line"}]
-    (is (= got expected))))
+(deftest add-original-wrapper-test
+  (let [segment {:some :values}
+        got (add-original-wrapper segment)]
+    (is (= got {:original {:some :values}}))))
 
 (deftest prepare-rows-test
   (let [got (prepare-rows {"line" "this is a log line"})
         expected {:rows [{"line" "this is a log line"}]}]
     (is (= got expected))))
+
+(defn raw-example
+  [kind]
+  (kwify-map (json/parse-stream (bs/to-reader (io/file (io/resource (str "test/example_" (name kind) ".json")))) true)))
```

Comment:
This is one where threading macros might help a bit:

```
(-> (str "test/example_" (name kind) ".json")
    io/resource
    io/file
    bs/to-reader
    (json/parse-stream true)
    kwify-map)
```

... which incidentally highlights an optimization, because io has a `reader` fn, which takes e.g. a URI (as returned by `resource`) and produces the reader you want, so you don't need to involve byte-streams. So that gets you:

```
(-> (str "test/example_" (name kind) ".json")
    io/resource
    io/reader
    (json/parse-stream true)
    kwify-map)
```

I'm not sure if you can just pass the filename to reader; you _might_ be able to. The documentation suggetss that it will try to open a string as a URL first, and if that fails, as a local file; I'm not sure if that means "local to cwd" or "follow Java classpath rules as resource does"; I believe it to be the former.